apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -78,56 +78,72 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
         }
 
         // Reintroduce color (Connelly: use eps to avoid scaling up noise w/ apollo3.png input)
-        Func color;
+        Func color("color");
         float eps = 0.01f;
         color(x, y, c) = outGPyramid[0](x, y) * (floating(x, y, c)+eps) / (gray(x, y)+eps);
 
         Func output("local_laplacian");
         // Convert back to 16-bit
         output(x, y, c) = cast<uint16_t>(clamp(color(x, y, c), 0.0f, 1.0f) * 65535.0f);
 
-
-
         /* THE SCHEDULE */
-        remap.compute_root();
-
-        if (get_target().has_gpu_feature()) {
-            // gpu schedule
-            Var xi, yi;
-            output.compute_root().gpu_tile(x, y, xi, yi, 16, 8);
-            for (int j = 0; j < J; j++) {
-                int blockw = 16, blockh = 8;
-                if (j > 3) {
-                    blockw = 2;
-                    blockh = 2;
+
+        if (!auto_schedule) {
```

Comment:
This if/else structure seems weird. Would be better more like the other apps:

    if (autoschedule) {
    } else if (gpu) {
    } else {
    }